drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,350 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+		}
+
+		// wait until there is only one pod in the running state
+		//TODO https://purestorage.atlassian.net/browse/PTX-23166 - This is a temporary fix to get the pod of the VM
+		testPod, err := GetVirtLauncherPodForVM(virtualMachineCtx, vol[0])
+		if err != nil {
+			return "", true, err
+		}
+
+		//Get the node where the vm is scheduled after the migration
+		nodeNameAfterMigration := testPod.Spec.NodeName
+
+		// Adding an extra check which keeps on checking for 5 mins if the pod has been migrated to another node (because in an overloaded cluster it might take time for the pod to be scheduled on another node)
+		podMigrated := func() (interface{}, bool, error) {
+			testPod, err := GetVirtLauncherPodForVM(virtualMachineCtx, vol[0])
+			if err != nil {
+				return "", true, err
+			}
+			nodeNameAfterMigration = testPod.Spec.NodeName
+			if nodeName == nodeNameAfterMigration {
+				return "", true, fmt.Errorf("VM pod not migrated to another node after migration triggered for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+			}
+			log.InfoD("VM pod migrated to node: [%s]", nodeNameAfterMigration)
+			return "", false, nil
+		}
+		_, err = task.DoRetryWithTimeout(podMigrated, 5*time.Minute, 30*time.Second)
+		if err != nil {
+			return "", false, err
+		}
+
+		if nodeName == nodeNameAfterMigration {
+			return "", false, fmt.Errorf("VM pod live migrated [%s] in namespace [%s] but is still on the same node [%s]", testPod.Name, testPod.Namespace, nodeName)
+		}
+		log.InfoD("VM pod live migrated to node: [%s]", nodeNameAfterMigration)
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetVirtLauncherPodForVM returns the virt-launcher pod for the VM
+func GetVirtLauncherPodForVM(virtualMachineCtx *scheduler.Context, vol *volume.Volume) (*corev1.Pod, error) {
+	pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pods for volume %s of context %s: %w", vol.ID, virtualMachineCtx.App.Key, err)
+	}
+
+	var found corev1.Pod
+	for _, pod := range pods {
+		if pod.Labels["kubevirt.io"] == "virt-launcher" && pod.Status.Phase == corev1.PodRunning {
+			if found.Name != "" {
+				// there should be only one VM pod in the running state (otherwise live migration is in progress)
+				return nil, fmt.Errorf("more than 1 KubeVirt pods (%s, %s) are in running state for volume %s",
+					found.Name, pod.Name, vol.ID)
+			}
+			found = pod
+		}
+	}
+	if found.Name == "" {
+		return nil, fmt.Errorf("failed to find a running pod for volume %s", vol.ID)
+	}
+	return &found, nil
+}
+
+// IsVMBindMounted checks if the volumes are bind mounted to the VM
+func IsVMBindMounted(virtualMachineCtx *scheduler.Context, driver scheduler.Driver, volumeDriver volume.Driver, initialCheck bool) (bool, error) {
```

Comment:
nit: rename initialCheck -> wait since that parameter is used to decide whether to wait for the bind mount or not

Reply:
Done.